benchmarks/harness/graphforge_bench/progressive_run.py

```
@@ -285,6 +287,322 @@ def _run_benchexec(stage: Path, executables: Executables, identities: Mapping[st
+            ):
+                candidates.append(value)
+    if len(candidates) != 1:
+        raise ControllerError("exact GraphForge certification evidence is missing or ambiguous")
+    return candidates[0]
+
+
+def _parse_benchexec_xml(raw_output: Path, *, correctness: bool) -> Mapping[str, Any]:
+    documents = sorted(raw_output.glob("*.xml"))
+    if len(documents) != 1:
+        raise ControllerError("exact BenchExec result XML is missing or ambiguous")
+    root = ET.parse(documents[0]).getroot()
```

Ruff (0.16.2)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)